src/components/Provider.js

```diff
@@ -20,7 +20,7 @@ function warnAboutReceivingStore() {
 
 export default class Provider extends Component {
   getChildContext() {
-    return { store: this.store }
+    return { store: this.store, react15CompatibilityMode: this.props.react15CompatibilityMode }
```

Comment:
This shouldn't be on `<Provider>`. It's on a per-component basis, like `pure`. I wouldn't worry about context at all.